sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDynamicPruningExpression.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import ai.rapids.cudf.ColumnVector
+
+import org.apache.spark.sql.catalyst.expressions.{DynamicPruning, Expression}
+
+case class GpuDynamicPruningExpression(child: Expression)
```

Comment:
Would we ever want this to return a scalar value? Like we do for `GpuKnownNotNull` and `GpuKnownFloatingPointNormalized`?

Reply:
Nice catch!  Fixed.